### PR TITLE
Docker: Support pre-installed plugins from other sources in custom Dockerfiles

### LINKED ISSUE #31302: Feature Request: Install plugins from other sources on custom Dockerfile.
**What would you like to be added**:
Add ability to install plugins from other sources via GF_INSTALL_PLUGINS when building a docker image.
**Why is this needed**:
You can make the ready-to-use Docker image with plugins from other sources.

For example, I want a Docker image with a Monaca data source plugin.
I can build a Docker image with this command

```bash
docker build \
  --build-arg "GRAFANA_VERSION=7.4.1" \
  --build-arg "GF_INSTALL_PLUGINS=https://github.com/openstack/monasca-grafana-datasource/archive/1.2.1.zip;monasca-grafana-datasource,grafana-clock-panel" \
  -t grafana-custom -f Dockerfile .
```

Refer to this document https://grafana.com/docs/grafana/latest/installation/docker/#install-plugins-from-other-sources. 
The GF_INSTALL_PLUGINS environment variable is supported to install plugins from other sources but it only supported when container startup.

Refer to this document https://grafana.com/docs/grafana/latest/installation/docker/#build-with-pre-installed-plugins. 
The GF_INSTALL_PLUGINS build arguments are only supported to install plugins from officials Grafana plugins source.

### PR DESCRIPTION
**Pull request by sgnsys3**
**What this PR does / why we need it**:

Refer to this document: [https://grafana.com/docs/grafana/latest/installation/docker/#build-and-run-a-docker-image-with-pre-installed-plugins](https://grafana.com/docs/grafana/latest/installation/docker/#build-and-run-a-docker-image-with-pre-installed-plugins)

Installing plugins from URL is not supported when building Docker image with pre-installed plugins.

**Which issue(s) this PR fixes**:
Add install plugins from URL script when building Docker image with pre-installed plugins.

Fixes: #31302

### CODE REVIEW

**Review by aknuds1 — COMMENTED** [review]
AFAICT, the technical changes make sense, but I'd have to hear with others to figure out if we want this. I think also documentation would have to be updated to reflect this support?

**Comment by marefr** [comment]
@sgnsys3 could you please open a feature request describing the use case more detailed and why you cannot customize this in your custom built image and why it should be included in our custom image. Then reference the created issue in the PR description. Thanks

**Comment by sgnsys3** [comment]
@marefr I have created and reference issue in this PR. This is correct?

**Review by marefr — CHANGES_REQUESTED** [review]
Based on discussion in the issue this looks good. But the ubuntu.Dockerfile needs update as well.

**Comment by sgnsys3** [comment]
@marefr I updated ubuntu.Dockerfile.

**Review by marefr — APPROVED** [review]
LGTM

**Review by marefr — CHANGES_REQUESTED** [review]
Was a bit fast there. 
- Please merge with master to resolve CI build problems.
- Please update this [section in the documentation](https://grafana.com/docs/grafana/latest/installation/docker/#build-with-pre-installed-plugins) to denote installing plugins from other sources is supported. For reference, here's the [markdown file](https://github.com/grafana/grafana/blob/master/docs/sources/installation/docker.md#build-with-pre-installed-plugins) that needs update.

File: docs/sources/installation/docker.md [hunk 1]
```diff
@@ -112,6 +112,19 @@ docker build \
 docker run -d -p 3000:3000 --name=grafana grafana-custom
 ```
 
+### Build with pre-installed plugins from other sources
+You can build a Docker image with plugins from other sources by specifying the URL like this: `GF_INSTALL_PLUGINS=<url to plugin zip>;<plugin name>`.
```
Comment (oddlittlebird) [inline docs/sources/installation/docker.md:116]:
```suggestion

You can build a Docker image with plugins from other sources by specifying the URL like this: `GF_INSTALL_PLUGINS=<url to plugin zip>;<plugin name>`.
```

**Review by achatterjee-grafana — APPROVED** [review]

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Comment by marefr** [comment]
Shouldn't need backport since [documentation](https://grafana.com/docs/grafana/latest/installation/docker/#build-and-run-a-docker-image-with-pre-installed-plugins) refers to this repository for building custom Docker image and the master branch is the default.

**Pull request merged**